### PR TITLE
Experience/8943/mdx general layout

### LINKED ISSUE #8943: Add generalizable sidenav/content MDX layout
## Problem statement

With MDX support, we'll be able to generalize our documentation pages through reusable layouts rather than the existing IATemplate logic.

## What you need to know

- This is intended to replace the existing [IATemplate logic](https://github.com/CDCgov/prime-reportstream/blob/master/frontend-react/src/components/Content/Templates/IATemplate.tsx#L24-L63) in the future

## Acceptance criteria

- [ ] There are MDX templates for sidebar navigation and page content
    - [ ] Nice-to-have: The templates are generalizable for any content page

## Follow-ups
- Convert existing Markdown pages to MDX
    - [#8952](https://app.zenhub.com/workspaces/web-experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8952)
    - [#8953](https://app.zenhub.com/workspaces/web-experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8953)

### PR DESCRIPTION
**Pull request by jpandersen87**
Closes #8943

This PR:
- adds a MarkdownLayout component for use with MDX components
  - It accepts the main MDX component article by natural component composition and an optional sidenav article via the sidenav prop
  - The layout uses a uswds grid-row to align the two articles side-by-side
- moves the internal/external link logic from the MarkdownRSLink component into a USSmartLink component that can be used by both the older markdown implementation and the new MarkdownLayout component
- adds a SideNavItem component
  - Provides the parent navlink and children sidenav element as siblings for use in the items prop of SideNav
  - Allows the children subnav to be visible when that path is active by default
- Updates USNavLink logic to include path hash in determining being active (/foo#bar and /foo will be considered active but not /foo#other vs current behavior)

### CODE REVIEW

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:16]: 🙌 🙌 🙌

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:24]: Should this take up a set number of columns rather than auto?  For example, the Product pages use four columns for the side nav and eight columns for the main content on tablet+ and just stack on smaller viewports.

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:29]: Same point here about taking up a set number of columns.

File: frontend-react/src/components/Content/MarkdownRSLink.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownRSLink.tsx:26]: Can we just remove this component entirely since it's just an alias to USSmartLink now?

File: frontend-react/src/components/Grid.tsx
Comment (stephenkao) [inline frontend-react/src/components/Grid.tsx:9]: Isn't this functionality already provided by [the Grid component](https://github.com/trussworks/react-uswds/blob/main/src/components/grid/Grid/Grid.tsx#L61)?

```tsx
<Grid className={ className } row>
    { children }
</Grid>
```

File: frontend-react/src/__mocks__/mdxjsReactMock.tsx
Comment (stephenkao) [inline frontend-react/src/__mocks__/mdxjsReactMock.tsx:6]: Simple simple 🙏

File: frontend-react/src/components/SideNav.test.tsx
Comment (stephenkao) [inline frontend-react/src/components/SideNav.test.tsx:10]: This feels fragile since it's so tightly coupled to an exact element hierarchy.  I'd recommend one of the following:

- Omit the entire [subnav](https://github.com/CDCgov/prime-reportstream/pull/8954/files#diff-0b6f9b990aac3ec98d4b3517114cacc1b3fff66b869546a5b049046ad5947be0R50-R54) OR add an `aria-hidden`/`hidden` if `isSubnavVisible` is `true`.  This would result in the query returning nothing for this link so we could run an expectation on that instead.
- Iterate upward through `parentElement`s until we find an ancestor element with a `display-none` class (since that should affect all descendants).  The other option is more preferable in my opinion since it's less reliant on a modifier class, though.

File: frontend-react/src/components/SideNav.tsx
Comment (stephenkao) [inline frontend-react/src/components/SideNav.tsx:44]: Honestly, I don't think it's a huuuuuge deal if we just want to build our own since [their SideNav](https://github.com/trussworks/react-uswds/blob/main/src/components/SideNav/SideNav.tsx) is so simplistic anyway.  Let's hold off on that and reconsider if it becomes an issue going forward, though!

File: frontend-react/src/components/SideNav.tsx
Comment (stephenkao) [inline frontend-react/src/components/SideNav.tsx:26]: Can we `export default` here?  I'm thinking there should always be a default export in component files to encourage us to have a 1:1 between files and components.  There'll be times that we'll have "sub-components" of course, but that's something we should try to avoid whenever possible.

File: frontend-react/src/components/SideNav.tsx
Comment (stephenkao) [inline frontend-react/src/components/SideNav.tsx:1]: Can we add this file to a new `src/shared` directory for common components?  cc @etanb who's doing a similar thing with the new Table component.

File: frontend-react/src/components/USLink.tsx
Comment (stephenkao) [inline frontend-react/src/components/USLink.tsx:219]: 🤓

**Review by stephenkao — CHANGES_REQUESTED** [review]
Added some points that I think should be addressed before merging, but this is looking great overall!

File: frontend-react/src/components/Grid.tsx
Comment (jpandersen87) [reply frontend-react/src/components/Grid.tsx:9]: Could be! If so, will drop this in favor of it. Thanks for the heads up!

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (jpandersen87) [reply frontend-react/src/components/Content/MarkdownLayout.tsx:56]: I was expecting that the usage of GridContainer here to be temporary until it's implemented further up the render tree.

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:24]: 🧑‍🎨

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:47]: 🙏

File: frontend-react/src/components/Content/MarkdownLayout.tsx
Comment (stephenkao) [inline frontend-react/src/components/Content/MarkdownLayout.tsx:61]: 🙏

**Review by stephenkao — APPROVED** [review]
Looks great to me!  Thanks for setting this up!

**Review by penny-lischer — APPROVED** [review]
LGTM

**Pull request merged**
